azure-pipelines.yml

```diff
@@ -227,6 +227,30 @@ jobs:
     condition: succeeded()
     workingDirectory: '$(System.DefaultWorkingDirectory)/tools/DependencyUpdater'
 
+  - task: DotNetCoreCLI@2
+    condition: succeeded()
+    displayName: 'Tests - Unit'
+    inputs:
+      command: 'test'
+      projects: |
+        **/*Tests.csproj
+        !**/obj/**
+        !**/bin/**/ref/**
+      arguments: '--configuration Release --collect "XPlat Code coverage"'
+      testRunTitle: 'Unit Tests'
```

Comment:
Up to you, but it might be better to continue on error here, so the unit tests are always uploaded, even if a test fails.

Reply:
Sure, I'll fix that later. But the build should break if any test fails.

Reply:
Yeah, the build will still show an error, but will also upload the results (so you can see which one). You could add a failing test to see what I mean...

Reply:
But you are right in the fact that the "build check" would pass, so might be a bad idea after all.

Reply:
How about running the report generator and publish result task regardless of whether the test project fails. I am not sure of the condition needed to do that though... @josesimoes , thoughts?

Reply:
Allow me to step in: 
- yes, the build HAS to fail on any Unit Test failing. 
- as soon as that occurs, the pipeline execution just stops, so there is no way to go to the "next" task and have it publish the coverage results
- and, to be honest, a Unit Test failing takes precedence on coverage. That can be looked upon after those pass. 😉

Reply:
perhaps `condition: always()`

Reply:
Nope.... that's the condition to run THIS task, so if the one before fails, no point on running Unit Test.

Reply:
I kinda meant like this: https://github.com/nanoframework/nanoFirmwareFlasher/pull/128/commits/6f3c156de891c8fdab3599ed8b0f2c41c62f758d

You will see that the tests fail (which they will on that PR, since they still need adding properly), so the build fails. But the test report is still uploaded and shows you where you need to look. But this can be added later, as not a priority.

Reply:
@networkfusion oh now I see what you're explaining. (I thought you were referring to the Unit Test task). Yes, that's OK to have those published even on test failing.